alf/algorithms/agent.py

```diff
@@ -186,6 +217,12 @@ def train_step(self, exp: Experience, state):
         info = AgentInfo()
         observation = self._encode(exp)
 
+        if self._goal_generator is not None:
+            new_state = new_state._replace(goal_generator=state.goal_generator)
+
+        if self._goal_generator is not None:
+            observation = [observation, state.goal_generator]
+
```

Comment:
Do you need also to call train_step for the goal generator (if it's trainable)?

Reply:
The current goal generator (RandomCategoricalGoalGenerator) is a non-trainable module.
Trainable goal generator will be done in a future PR.

Reply:
I think it's still better to leave a train_step entrance here. RandomCategoricalGoalGenerator can just pass this function.

Reply:
Train_step entrance added.